thorlcr/activities/fetch/thfetch.cpp

```diff
@@ -93,6 +100,32 @@ class CFetchActivityMaster : public CMasterActivity
         if (!container.queryLocalOrGrouped())
             dst.append((int)mpTag);
     }
+    virtual void deserializeStats(unsigned node, MemoryBuffer &mb)
+    {
+        CMasterActivity::deserializeStats(node, mb);
+        for (auto &stats: subFileStats)
+            stats->deserialize(node, mb);
+    }
+    virtual void done() override
+    {
+        if (!subFileStats.empty())
+        {
+            unsigned numSubFiles = subFileStats.size();
+            for (unsigned i=0; i<numSubFiles; i++)
+            {
+                IDistributedFile *file = queryReadFile(i);
+                if (file)
+                    file->addAttrValue("@numDiskReads", subFileStats[i]->getStatisticSum(StNumDiskReads));
+            }
+        }
+        else
+        {
+            IDistributedFile *file = queryReadFile(0);
+            if (file)
```

Comment:
not for this PR, but updateFileAttrs() in dadfs.cpp should probably aggregate the sub NumDiskReads and set the total into the super.

Reply:
I've created a jira to do this later https://track.hpccsystems.com/browse/HPCC-26262